src/VisualStudio/Core/Test/DocumentRefactoring/DocumentRefactoringServiceTests.vb

```diff
@@ -0,0 +1,107 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis.Test.Utilities
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces
+Imports Microsoft.CodeAnalysis.Host
+Imports Microsoft.CodeAnalysis.LanguageServices
+Imports Microsoft.CodeAnalysis.Shared.Extensions
+Imports Microsoft.CodeAnalysis.Editor.UnitTests
+Imports Microsoft.VisualStudio.LanguageServices.Implementation.DocumentRefactoring
+Imports Microsoft.VisualStudio.Composition
+Imports System.IO
+Imports System.Xml
+
+Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.DocumentRefactoring
+    <UseExportProvider>
+    Public Class DocumentRefactoringServiceTests
+        Private ReadOnly _catalog As ComposableCatalog = TestExportProvider.EntireAssemblyCatalogWithCSharpAndVisualBasic.WithPart(GetType(DocumentRefactoringService))
+        Private ReadOnly _factory As IExportProviderFactory = ExportProviderCache.GetOrCreateExportProviderFactory(_catalog)
+
+        Private Function TestDocumentRefactoring(markup As String, Optional expectMatchingName As Boolean = True) As Task
+            Dim workspace = TestWorkspace.CreateCSharp(markup, exportProvider:=_factory.CreateExportProvider())
+            Return TestDocumentRefactoring(workspace, expectMatchingName:=expectMatchingName)
+        End Function
+
+        Private Function TestDocumentRefactoring(files() As String, Optional ExpectMatchingName As Boolean = True) As Task
+            Dim workspace = TestWorkspace.CreateCSharp(files, exportProvider:=_factory.CreateExportProvider())
+            Return TestDocumentRefactoring(workspace, expectMatchingName:=ExpectMatchingName)
+        End Function
+
+        Private Async Function TestDocumentRefactoring(workspace As TestWorkspace, Optional expectMatchingName As Boolean = True) As Task
+            Dim project = workspace.CurrentSolution.Projects.Single()
+            Dim document = project.Documents.First()
+            Dim refactorService = workspace.GetService(Of DocumentRefactoringService)
+
+            Await refactorService.UpdateAfterInfoChangeAsync(document, document)
```

Comment:
So what is this actually doing then....?

Reply:
the wrong thing :)